src/main/java/org/sagebionetworks/web/client/GlobalApplicationStateImpl.java

```diff
@@ -618,13 +622,20 @@ public void gotoLoginPage() {
     Date twoHoursFromNow = new Date();
     twoHoursFromNow.setTime(twoHoursFromNow.getTime() + (2 * 60 * 60 * 1000));
 
+    String returnUrl = gwt.getCurrentURL();
+    String lastPlacePath = sessionStorage.getItem(
+      GlobalApplicationStateImpl.LAST_PLACE
+    );
+    if (getCurrentPlace() instanceof LoginPlace && lastPlacePath != null) {
+      returnUrl = gwt.getHostPageBaseURL() + lastPlacePath;
+    }
```

Comment:
if this is called from the LoginPlace, redirect to the last place (stored in sessionStorage)